main/ChecksBalancesMod.F90

```diff
@@ -240,7 +243,134 @@ subroutine CheckLitterPools(currentSite,bc_in)
+               ediag => site%flux_diags%elem(el), &
+               diag  => site%flux_diags, &
+               site_mass => site%mass_balance(el))
+
+            ! Initialize the integrated flux balance diagnostics
+            ! No need to initialize the instantaneous states, those are re-calculated
+
+            ibal%state_liveveg = (biomass_stock + seed_stock)*area_inv
+            ibal%state_litter  = litter_stock * area_inv
+
+            ! Flux for live veg: net uptake (either NPP or net nutrient uptake) +
+            !                    net spatial seed flux -
+            !                    veg turnover to litter -
+            !                    veg loss to fire (SEEDS DONT BURN) -
+            !                    veg loss from exported harvest - 
+            !                    seed turnover
+
+            tot_litter_input = (sum(ediag%leaf_litter_input(:)) + &
+                 sum(ediag%root_litter_input(:)) + &
+                 sum(ediag%cwd_ag_input(:)) + &
+                 sum(ediag%cwd_bg_input(:)))*area_inv
+
+            select case(element_list(el))
+            case(carbon12_element)
+               net_uptake = diag%npp + site_mass%net_root_uptake*area_inv
+            case(nitrogen_element)
+               net_uptake = site_mass%net_root_uptake*area_inv
+            case(phosphorus_element)
+               net_uptake = site_mass%net_root_uptake*area_inv
+            case default
+               write(fates_log(),*) 'FATES: Invalid choice for cohort_fusion_conservation_method'
+               call endrun(msg=errMsg(sourcefile, __LINE__))
+            end select
 
+            ! Fluxes are in [kg/m2/day] so they can be added,
+            ! the frequency is /day so no conversion necessary
+            ! to integrate to [kg/m2]
+            ibal%iflux_liveveg = ibal%iflux_liveveg + &
+                 ( net_uptake          &
+                 - tot_litter_input    &
+                 - ediag%burned_liveveg &
+                 - ediag%exported_harvest & 
+                 + site_mass%seed_in*area_inv &
+                 - site_mass%seed_out*area_inv &
+                 - ediag%tot_seed_turnover)
+
+            ! Flux for litter: veg turnover + seed turnover - "these are tot_litter_input"
+            !                  fragmentation - 
+            !                  burned litter
+
+            ibal%iflux_litter = ibal%iflux_litter + &
+                 tot_litter_input - &
+                 (site_mass%frag_out*area_inv - ediag%tot_seed_turnover) - &
+                 (site_mass%burn_flux_to_atm*area_inv - ediag%burned_liveveg)
+
+
+            ediag%err_liveveg = ibal%iflux_liveveg - ibal%state_liveveg
+
+            ediag%err_litter  = ibal%iflux_litter - ibal%state_litter
+
+
+            
+            !print*, ediag%err_liveveg!, ediag%err_liveveg/ibal%state_liveveg, ibal%state_liveveg, net_uptake, tot_litter_input
+
+            ! Perform the comparison between integrated flux and state
```

Comment:
I think we discussed this but not sure where we landed on having this turned on or not. Could it be an option so a user could opt in to having their run crash if states and integrated fluxes start to diverge?

Reply:
I added an issue here: https://github.com/NGEET/fates/issues/1264